src/features/tutor/tutor.service.js

```diff
@@ -27,25 +28,70 @@ exports.getTutors = async ({
   approvalStatus = "approved",
   profileVisibility = "active",
   subjects,
-  availability,
+  name,
+  ratings,
   limit = 10,
   page = 1,
 }) => {
+  const includes = [];
+  let where = {
+    approvalStatus,
+    profileVisibility,
+  };
+
+  //Subject
   const subjectInclude = {
-    model: Subject,
+    model: Subject.scope("join"),
     as: "subjects",
   };
   if (subjects && subjects.length > 0) {
-    subjectInclude.query = {
-      [Op.in]: subjects,
+    subjectInclude.where = {
+      id: { [Op.in]: subjects.map(Number) },
+    };
+  }
+
+  includes.push(subjectInclude);
+
+  //Name
+  if (name) {
+    const searchWords = name.split(" ");
+
+    const conditions = searchWords.map((word) => ({
+      [Op.or]: [
+        { first_name: { [Op.iLike]: `%${word}%` } },
+        { last_name: { [Op.iLike]: `%${word}%` } },
+      ],
+    }));
+
+    const nameInclude = {
+      model: User.scope("join"),
+      as: "user",
+      where: conditions,
     };
+    includes.push(nameInclude);
+  }
+
+  //Ratings
+  //   const where = {
+  //   status: 'active',
+  //   funding: 'funded',
+  // };
+
+  // await sequelize.query(sql`SELECT * FROM projects WHERE ${sql.where(where)}`);
+  if (ratings && ratings.length > 0) {
+    const ratingWhere = sequelize.where(
+      sequelize.fn("ROUND", sequelize.col("rating")),
+      {
+        [Op.in]: ratings,
+      }
+    );
+    where = { [Op.and]: [{profileVisibility}, {approvalStatus}, ratingWhere] };
+    console.log(ratingWhere);
   }
+
   return await Tutor.scope("join").findAndCountAll({
```

Comment:
Right now, the total value in meta is counting all tutors in the database, not just those with the specified conditions, approvalStatus and profileVisibility.
<img width="943" height="333" alt="image" src="https://github.com/user-attachments/assets/da213baf-48e0-425a-b43f-1d7e059d87f6" />

This happens because Sequelize’s findAndCountAll doesn’t automatically apply filters across joined tables unless distinct: true is used.
Suggestion: Add distinct: true to the query options, e.g.:

```suggestion
return await Tutor.scope("join").findAndCountAll({
  where,
  include: includes,
  limit,
  offset: (page - 1) * limit,
  distinct: true, // ensures total count matches filtered rows
});
```